### PR TITLE
Lizeth

### PR DESCRIPTION
**Pull request by lvelascom**

### CODE REVIEW

File: advisors/models.py [hunk 1]
```diff
@@ -3,14 +3,44 @@
...
+    email = models.CharField(max_length=256)
+    zip = models.IntegerField()
+    title = models.CharField(max_length=10),
+
+
+class Group(models.Model):
+    name = models.CharField(max_length=128)
+    archived = models.BooleanField()
+    deleted = models.BooleanField()
+    code = models.CharField(max_length=140)
```
Comment (lufte) [inline advisors/models.py:24]: This model is the same as studentgroup. `archived`, `deleted` and `code` fields should be moved to the other model and this model should be removed.

File: advisors/models.py [hunk 2]
```diff
@@ -3,14 +3,44 @@
 
 class StudentGroup(models.Model):
     name = models.CharField(max_length=128)
-    students = models.ManyToMany('students.Student')
+    students = models.ManyToManyField('students.Student')
```
Comment (lufte) [inline advisors/models.py:6]: Ideally we should always create the model for the many-to-many relation.  Check https://docs.djangoproject.com/en/dev/ref/models/fields/#django.db.models.ManyToManyField.through

File: advisors/models.py [hunk 3]
```diff
@@ -3,14 +3,44 @@
...
 class Advisor(models.Model):
-    pass
+    role = models.CharField(max_length=128)
+    username = models.CharField(max_length=128)
+    first_name = models.CharField(max_length=128)
+    last_name = models.CharField(max_length=128)
+    phone_number = models.CharField(max_length=16)
+    email = models.CharField(max_length=256)
+    zip = models.IntegerField()
+    title = models.CharField(max_length=10),
```
Comment (lufte) [inline advisors/models.py:17]: Add a ManyToMany field using AdvisorGroup as the `through` model.

File: advisors/models.py [hunk 4]
```diff
@@ -3,14 +3,44 @@
...
+    class Meta:
+        unique_together = ('advisor', 'group')
 
 
 class Organization(models.Model):
     name = models.CharField(max_length=128)
-    groups = models.ManyToMany(Group)
-    advisors = models.ManyToMany(Advisor)
+    groups = models.ManyToManyField(Group)
+    advisors = models.ManyToManyField(Advisor)
```
Comment (lufte) [inline advisors/models.py:38]: Add the `through` attributes here too.

File: advisors/models.py [hunk 5]
```diff
@@ -3,14 +3,44 @@
...
+
+
+class GroupStudent(models.Model):
+    student = models.ForeignKey('students.Student', on_delete=models.CASCADE)
+    group = models.ForeignKey(Advisor, on_delete=models.CASCADE)
```
Comment (lufte) [inline advisors/models.py:43]: This foreign key is called group but points to Advisor.

File: students/models.py [hunk 6]
```diff
@@ -1,18 +1,35 @@
...
+
+ETHNICITY_CHOICES = [
+    ('American Indian and Alaska Native', 'American Indian and Alaska Native'),
+    ('Asian', 'Asian'),
+    ('Black or African American', 'Black or African American'),
+    ('Hispanic', 'Hispanic'),
+    ('Native Hawaiian and Pacific Islander',
+     'Native Hawaiian and Pacific Islander'),
+    ('White', 'White')
+]
```
Comment (lufte) [inline students/models.py:14]: Follow the guidelines here (https://docs.djangoproject.com/en/dev/ref/models/fields/#django.db.models.Field.choices) for definig the choices of a field. Even better, we could use the new django constraints (https://docs.djangoproject.com/en/3.1/ref/models/constraints/#check) so this check is done at a RDBMS level.

**Review by lufte — CHANGES_REQUESTED** [review]

**Review by lufte — APPROVED** [review]

**Review by jjcp — APPROVED** [review]

**Pull request merged**